curvefs/src/client/rpcclient/metaserver_client.cpp

```diff
@@ -539,14 +651,104 @@ MetaStatusCode MetaServerClientImpl::GetOrModifyS3ChunkInfo(
 
     auto taskCtx = std::make_shared<TaskContext>(
         MetaServerOpType::GetOrModifyS3ChunkInfo, task, fsId, inodeId);
-    GetOrModifyS3ChunkInfoExcutor excutor(opt_, metaCache_, channelManager_);
-    return ReturnError(excutor.DoRPCTask(taskCtx));
+    GetOrModifyS3ChunkInfoExcutor excutor(
+        opt_, metaCache_, channelManager_, taskCtx);
+    return ConvertToMetaStatusCode(excutor.DoRPCTask());
+}
+
+class GetOrModifyS3ChunkInfoRpcDone : public MetaServerClientRpcDoneBase {
+ public:
+    GetOrModifyS3ChunkInfoRpcDone(TaskExecutorDone *done,
+        const std::shared_ptr<MetaServerClientMetric> &metaserverClientMetric):
+        MetaServerClientRpcDoneBase(done, metaserverClientMetric) {}
+
+    virtual ~GetOrModifyS3ChunkInfoRpcDone() {}
+    void Run() override;
+    GetOrModifyS3ChunkInfoResponse response;
+};
+
+void GetOrModifyS3ChunkInfoRpcDone::Run() {
+    std::unique_ptr<GetOrModifyS3ChunkInfoRpcDone> self_guard(this);
+    brpc::ClosureGuard done_guard(done_);
+    auto taskCtx = done_->GetTaskExcutor()->GetTaskCxt();
+    auto& cntl = taskCtx->cntl_;
+    auto metaCache = done_->GetTaskExcutor()->GetMetaCache();
+    if (cntl.Failed()) {
+        metaserverClientMetric_->appendS3ChunkInfo.eps.count << 1;
+        LOG(WARNING) << "GetOrModifyS3ChunkInfo Failed, errorcode: "
+                     << cntl.ErrorCode()
+                     << ", error content: " << cntl.ErrorText()
+                     << ", log id: " << cntl.log_id();
+        done_->SetRetCode(-cntl.ErrorCode());
+        return;
+    }
+
+    MetaStatusCode ret = response.statuscode();
+    if (ret != MetaStatusCode::OK) {
+        LOG(WARNING) << "GetOrModifyS3ChunkInfo, inodeId: "
+                     << taskCtx->inodeID
+                     << ", fsId: " << taskCtx->fsID
+                     << ", errorcode: " << ret
+                     << ", errmsg: " << MetaStatusCode_Name(ret);
+        done_->SetRetCode(ret);
+        return;
+    } else if (response.has_appliedindex()) {
```

Comment:
why not judge s3ChunkInfoMap in GetOrModifyS3ChunkInfoResponse?

Reply:
judge what? it can be empty.